macos/FluentUI/Button.swift

```diff
@@ -0,0 +1,241 @@
+//
+//  Copyright (c) Microsoft Corporation. All rights reserved.
+//  Licensed under the MIT License.
+//
+
+import AppKit
+
+/// Indicates what style our button is drawn as
+@objc public enum ButtonStyle: Int, CaseIterable {
+    case primaryFilled	// Solid fill color
+    case primaryOutline	// Clear fill color, solid outline
+    case borderless		// Clear fill color, clear outline
+}
+
+// MARK: - Button
+
+/// A fluent styled button, with hover effects and a corner radius.
+@objc(MSFButton)
+open class Button: NSButton {
+	/// Initializes a Fluent UI Button with a title, automatically colored to suit the style
+	/// - Parameters:
+	///   - title: String displayed in the button
+	///   - style: The ButtonStyle, defaulting to primaryFilled
+	@objc public init(title: String, style: ButtonStyle = .primaryFilled) {
+		super.init(frame: .zero)
+		initialize(title: title, image: nil, style: style)
+    }
+	
+	/// Initializes a Fluent UI Button with an image, setting the imagePosition to imageOnly
+	/// - Parameters:
+	///   - image: The NSImage to diplay in the button
+	///   - style: The ButtonStyle, defaulting to primaryFilled
+	@objc public init(image: NSImage, style: ButtonStyle = .primaryFilled) {
+		super.init(frame: .zero)
+		initialize(title: nil, image: image, style: style)
+    }
+
+	@available(*, unavailable)
+	required public init?(coder decoder: NSCoder) {
+		preconditionFailure()
+	}
+	
+	private func initialize(title: String?, image: NSImage?, style: ButtonStyle = .primaryFilled) {
+		// Do common initialization work
+		isBordered = false
+		wantsLayer = true
+		
+		self.style = style
+		
+		if let title = title {
+			self.title = title
+		}
+		
+		if let image = image {
+			self.image = image
+		}
+	}
+	
+	override public var image: NSImage? {
+		willSet {
+			guard wantsLayer == true else {
+				exit(-1)
+			}
+			imagePosition = .imageOnly
+		}
+	}
+	
+	override public var title: String {
+		willSet {
+			guard wantsLayer == true else {
+				exit(-1)
```

Comment:
Would prefer a preconditionFailure than an exit(-1) since this seems like exactly what that would be used for